modules/WebPublishPlugin/src/main/java/net/clementlevallois/web/publish/plugin/controller/PublishingActions.java

```diff
@@ -0,0 +1,147 @@
+        exporterGexf.setWorkspace(workspace);
+        exporterGexf.setExportDynamic(false);
+
+        StringWriter stringWriter = new StringWriter();
+        ec.exportWriter(stringWriter, exporterGexf);
+        String gexfToSendAsString = stringWriter.toString();
+
+        int sizeGexfInBytes = gexfToSendAsString.getBytes(StandardCharsets.UTF_8).length; //
+
+        // Files pushed to a github shouldn't be larger than 100Mb
+        // There is still a doubt: is this limit the same for uploads to a gist?
+        if (sizeGexfInBytes >= (BYTES_IN_A_MEGABYTE * MAX_MB_FOR_GITHUB_PUSH)) {
+            jsonObject.addProperty(ERROR_CODE_GEXF_TOO_BIG, bundle.getString("general.message.error.network_too_big"));
+            return jsonObject;
+        }
+
+        if (graph.getNodeCount() > 10_000 || graph.getEdgeCount() > 20_000) {
+            jsonObject.addProperty(SUCCESS_BUT_WITH_WARNING, gexfToSendAsString);
+        } else {
+            jsonObject.addProperty(SUCCESS_CODE, gexfToSendAsString);
+        }
+        return jsonObject;
+    }
+
+    public static JsonObject connectToGithub() {
+        JsonObject responseCOnnectGithubAsJO = null;
+        try {
+            HttpClient client = HttpClient.newHttpClient();
+            String inputParams = "client_id=" + GEPHI_APP_CLIENT_ID + "&scope=gist";
+            String url = "https://github.com/login/device/code";
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .header("accept", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(inputParams))
+                    .build();
+
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            JsonElement responseAsJsonElement = JsonParser.parseString(response.body());
+            responseCOnnectGithubAsJO = responseAsJsonElement.getAsJsonObject();
+
+        } catch (IOException | InterruptedException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+        return responseCOnnectGithubAsJO;
+    }
+
+    public static JsonObject postGexfToGist(String gexfFile, String access_token, String fileName) {
+        JsonObject responsePostGexfToGist = new JsonObject();
+        JsonObject bodyPostGexfToGist = new JsonObject();
+        bodyPostGexfToGist.addProperty("description", bundle.getString("general.message.file_sent_from_gephi"));
+        bodyPostGexfToGist.addProperty("public", "true");
+        JsonObject fileItself = new JsonObject();
+        JsonObject contentsFile = new JsonObject();
+        contentsFile.addProperty("content", gexfFile);
+        fileItself.add(fileName, contentsFile);
+        bodyPostGexfToGist.add("files", fileItself);
+        String bodyToString = bodyPostGexfToGist.toString();
+
+        try {
+            HttpClient client = HttpClient.newHttpClient();
+            String url = "https://api.github.com/gists";
```

Comment:
Hardcoded URL

Reply:
If the URL ever changes, the query parameters might change as well. For this reason, I would rather keep in the same view / location both the URL and the query params.